### ISSUE TITLE
US1 - PiRogue VPN

### ISSUE THREAD

**Issue by U039b** [PiRogueToolSuite/project-management#2]
**Description:** The PiRogue VPN project aims to facilitate the operational effectiveness of organizations in assisting individuals at risk. By deploying PiRogue as [an emergency VPN](https://www.civilsphereproject.org/emergency-vpn) server, people at risk can easily connect their device and security analysts can analyze the network traffic of the device. Alerts and DPI data will be sent to Colander for further analysis, and the management of both virtual and physical PiRogues will be centralized through Colander. This activity mainly consists in integrating [Wireguard](https://www.wireguard.com/). This feature will allow users at risk who do not have a physical PiRogue to get their phone’s network traffic analyzed by any organization running a PiRogue VPN service. The end-user simply has to install the Wireguard app on their device and flash the QR-code provided by the organization running the service.

**Tasks:** 
- [ ] PiRogueToolSuite/project-management#3 
- [ ] PiRogueToolSuite/project-management#4 
- [ ] PiRogueToolSuite/project-management#5
- [ ] PiRogueToolSuite/project-management#6

**Comment by CyrilBrulebois** [comment]
#### This month

The initial integration of WireGuard into the `pirogue-networking` and `pirogue-admin` packages happened in August. The `pirogue-networking` side was rather easy as everything was already meant to be as generic as possible in that package, while `pirogue-admin` needed several updates. On one hand, the existing `pirogue-admin` CLI tool was extended to support suggesting VPN mode when autodetection is requested (that happens if the system only reports a single network interface). On the other hand, a new `pirogue-admin-wireguard` CLI tool was introduced to manage all WireGuard-related aspects separately; this includes generating configuration files and/or QR codes to configure the WireGuard application on the target phones.

#### Next month

A few issues were opened to keep track of proposed improvements and cleanup, but VPN support is considered to be feature-complete already.

#### Challenges

None.

**Comment by CyrilBrulebois** [comment]
#### This month

The `pirogue-admin` package was extended to support WireGuard: that means configuring the relevant service but also providing an API and a CLI to manage peers: adding/listing/deleting them, but also generating a configuration file or a QR code to automatically configure the WireGuard application on phones to be analyzed.

This was confirmed to work fine in a virtual machine accessible from the internet (wiring the relevant UDP port to the “ViRogue” virtual machine).

#### Next month

No more work is planned on this topic.

#### Challenges

The `pirogue-networking` package needed to be modified slightly to account for inter-service interactions that hadn't been problematic before.

**Comment by CyrilBrulebois** [comment]
Adding support for VPN mode is covered in the following issue: https://github.com/PiRogueToolSuite/project-management/issues/28

TL;DR: Initial VPN support added in August, with more tests in “public environments” (e.g. Hetzner) in September.
